src/ui/legend_ui.cpp

```diff
@@ -0,0 +1,61 @@
+#include "legend_ui.h"
+#include "events.h"
+
+// Legend presets
+static const LegendUI::str_pair BUILD_MENU_LEGEND = {
+        {"xboxControllerDPad.png", "Move selection"},
+        {"xboxControllerButtonA.png", "Choose"},
+        {"xboxControllerStart.png", "Ready up"},
+};
+static const LegendUI::str_pair BUILD_SELECTED_LEGEND = {
+        {"xboxControllerLeftThumbstick.png", "Select location"},
+        {"xboxControllerButtonA.png", "Build"},
+        {"xboxControllerButtonB.png", "Cancel"},
+};
+
+LegendUI::LegendUI(float aspect, float legend_width_, float entry_height_, float padding_) :
+        UI((100.f - (legend_width_ + padding_)) * aspect, 100.f),
```

Comment:
Do we need to set the root start coordinates here seeing as we do it in set_legend?